### PR TITLE
Add welcome screen page and component

### PR DESCRIPTION
**Pull request by tnightingale**
To do:
- [x] Tests for welcome page / actions
- [x] Wire up 'Get started' button to navigate to Explore

Issue #105 

I used the application icon I arbitrarily chose as the logo placeholder.

Missing (TBD if within scope of AC):
- the "is first time running application" persisted value
- the related "no language selected" state in the language picker

### CODE REVIEW

**Comment by tnightingale** [comment]
Tests passing, thanks @trisapeace!

File: src/components/welcome/welcome.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            {
+                flex: 1,
+                justifyContent: 'flex-start',
+                alignItems: 'center',
+                padding: 20,
+            },
+        ]}>
+            <Image
+                source={logoImg}
+                style={[
```
Comment (bensinjin) [inline src/components/welcome/welcome.tsx]: What's the array syntax adding? Isn't this the same as `style={{width: 200, height:200, marginTop: 50, marginBottom:20}}`?

Comment (bensinjin) [reply src/components/welcome/welcome.tsx]: I thought it was used to merge styles?

Comment (bensinjin) [reply src/components/welcome/welcome.tsx]: ie. 
```
const bgColor = {backgroundColor: "red"}
<View style{[{padding: 10}, bgColor]} />
```

Comment (tnightingale) [reply src/components/welcome/welcome.tsx]: Yep, I just separated them here based on logical (in my head) grouping as a half-step towards putting them external styles. The array syntax is nicer IMHO as it makes it easier to quickly add/remove styles.

**Comment by rasmus-storjohann-PG** [comment]
I'm fine with both raised issued being out of scope for this task

File: src/components/welcome/welcome.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    readonly availableLocales: ReadonlyArray<LocaleInfo>;
+}
+
+export interface Actions {
+    readonly setLocale: (localeCode: string) => SetLocale.Request;
+    readonly goToExplore: () => void;
+}
+
+// tslint:diable-next-line:no-var-requires
+// @ts-ignore: React Native uses require() to load ImageBitmaps; it's best to do this once, outside of render.
```
Comment (rasmus-storjohann-PG) [inline src/components/welcome/welcome.tsx]: Switch lines so that the next line tslint statement works

File: src/components/welcome/welcome.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import { LocaleInfo, Locale } from '../../locale';
+import { SetLocale } from '../../stores/locale';
+
+export interface Props {
+    readonly isFirstRun: boolean;
+    readonly currentLocale: Locale;
+    readonly availableLocales: ReadonlyArray<LocaleInfo>;
+}
+
+export interface Actions {
```
Comment (rasmus-storjohann-PG) [inline src/components/welcome/welcome.tsx]: Rename to `WelcomeProps` and `WelcomeActions`

File: src/components/welcome/welcome.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                            style={{ width: undefined }}
+                            placeholder={props.i18n.t`Select your language`}
+                            placeholderStyle={{ color: '#bfc6ea' }}
+                            placeholderIconColor='#007aff'
+                            selectedValue={props.currentLocale.code}
+                            onValueChange={props.setLocale}>
+                        {props.availableLocales.map((locale: LocaleInfo) => (
+                            <Picker.Item key={locale.code} label={locale.label} value={locale.code} />
+                        ))}
+                    </Picker>
```
Comment (rasmus-storjohann-PG) [inline src/components/welcome/welcome.tsx]: When I run this on my phone, I don't see the picker :-(

Comment (rasmus-storjohann-PG) [reply src/components/welcome/welcome.tsx]: Actually, this is not a minor nit, kinda a big deal actually

File: src/components/welcome/welcome.tsx (hunk 2 above)
Comment (tnightingale) [reply src/components/welcome/welcome.tsx]: Are you sure this makes a difference? It works for me either way...

File: src/components/welcome/welcome.tsx (hunk 3 above)
Comment (tnightingale) [reply src/components/welcome/welcome.tsx]: Done

File: src/components/welcome/welcome.tsx (hunk 4 above)
Comment (tnightingale) [reply src/components/welcome/welcome.tsx]: Weird, I'll take another look.

**Pull request merged**